modules/auxiliary/scanner/http/xst.rb

```diff
@@ -0,0 +1,64 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  # Exploit mixins should be called first
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::WmapScanServer
+  # Scanner mixin should be near last
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name'        => 'Cross-Site Tracing (XST) Checker',
+      'Description' => 'Checks if the host is vulnerable to Cross-Site Tracing (XST)',
+      'Author'       => [ 'Jay Turla <@shipcod3>' ],
+      'License'     => MSF_LICENSE
+    )
+  end
+
+  def run_host(target_host)
+
+    begin
+      res = send_request_raw({
+        'version'      => '1.0',
+        'uri'          => '/<script>alert("TRACE Payload");</script>', #XST Payload
+        'method'       => 'TRACE',
+        'headers' =>
+        {
+          'Cookie' => "TRACE ME PLS",
+        },
+      }, 10)
```

Comment:
No need to specify the timeout. The HTTPClientTimeout is user configurable and the default timeout is 20. If you still need it to be 10, you can register it as an advanced datastore option.